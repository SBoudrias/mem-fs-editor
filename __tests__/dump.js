'use strict';

const filesystem = require('fs');
const os = require('os');
const path = require('path');
const memFs = require('mem-fs');
const editor = require('..');

const rmSync = filesystem.rmSync || filesystem.rmdirSync;

describe('#dump()', () => {
  const output = path.join(os.tmpdir(), '/mem-fs-editor-test');
  const subdir = 'foo';
  let store;
  let fs;

  beforeEach(async () => {
    store = memFs.create();
    fs = editor.create(store);

    fs.write(path.join(output, subdir, 'committed'), 'committed');
    fs.write(path.join(output, subdir, 'committed-delete'), 'committed-delete');
    fs.write(path.join(output, subdir, 'committed-changed'), 'orig');
    fs.write(path.join(output, subdir, 'committed-delete-changed'), 'orig');
    await fs.commit();

    fs.delete(path.join(output, subdir, 'committed-delete'));
    await fs.commit();

    fs.write(path.join(output, subdir, 'committed-changed'), 'committed-changed');
    fs.write(path.join(output, subdir, 'not-committed'), 'not-committed');
    fs.write(path.join(output, subdir, 'not-committed-delete'), 'not-committed-delete');

    fs.delete(path.join(output, subdir, 'committed-delete-changed'));
    fs.delete(path.join(output, subdir, 'not-committed-delete'));
  });

  afterEach(() => {
    rmSync(output, { recursive: true, force: true });
  });

  it('should match snapshot', () => {
    expect(fs.dump(output)).toMatchSnapshot();
  });

  describe('with a glob pattern', () => {
    it('should return files that matches the pattern and have state or stateCleared', () => {
      expect(fs.dump(output, '**/*committed')).toMatchSnapshot();
    });
  });
});
