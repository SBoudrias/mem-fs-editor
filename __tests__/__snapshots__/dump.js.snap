// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#dump() should match snapshot 1`] = `
Object {
  "foo/committed": Object {
    "contents": "committed",
    "stateCleared": "modified",
  },
  "foo/committed-changed": Object {
    "contents": "committed-changed",
    "state": "modified",
    "stateCleared": "modified",
  },
  "foo/committed-delete": Object {
    "contents": null,
    "stateCleared": "deleted",
  },
  "foo/committed-delete-changed": Object {
    "contents": null,
    "state": "deleted",
    "stateCleared": "modified",
  },
  "foo/not-committed": Object {
    "contents": "not-committed",
    "state": "modified",
  },
  "foo/not-committed-delete": Object {
    "contents": null,
    "state": "deleted",
  },
}
`;

exports[`#dump() with a glob pattern should return files that matches the pattern and have state or stateCleared 1`] = `
Object {
  "foo/committed": Object {
    "contents": "committed",
    "stateCleared": "modified",
  },
  "foo/not-committed": Object {
    "contents": "not-committed",
    "state": "modified",
  },
}
`;
